src/location/locationService.ts

```diff
@@ -11,7 +11,8 @@ export class Ng2LocationServices extends BaseLocationServices {
   }
 
   _get() {
-    return this._locationStrategy.path(true);
+    return this._locationStrategy.path(true)
+      .replace(this._locationStrategy.getBaseHref().replace(/\/$/, ''), '');
```

Comment:
it's unfortunate the location strategy class doesn't return the path without the base href.
It knows how to generate urls with the base href via `prepareExternalUrl`.